cardano-testnet/src/Testnet/Components/Configuration.hs

```diff
@@ -95,30 +97,29 @@ createSPOGenesisAndFiles
   -> ConwayGenesis StandardCrypto -- ^ The conway genesis to use, for example 'Defaults.defaultConwayGenesis'.
   -> TmpAbsolutePath
   -> m FilePath -- ^ Shelley genesis directory
-createSPOGenesisAndFiles (NumPools numPoolNodes) era shelleyGenesis alonzoGenesis conwayGenesis (TmpAbsolutePath tempAbsPath) = do
-  let genesisShelleyFpAbs = tempAbsPath </> defaultGenesisFilepath ShelleyEra
-      genesisAlonzoFpAbs = tempAbsPath </> defaultGenesisFilepath AlonzoEra
-      genesisConwayFpAbs = tempAbsPath </> defaultGenesisFilepath ConwayEra
-      genesisShelleyDirAbs = takeDirectory genesisShelleyFpAbs
+createSPOGenesisAndFiles (NumPools numPoolNodes) era shelleyGenesis alonzoGenesis conwayGenesis (TmpAbsolutePath tempAbsPath) = GHC.withFrozenCallStack $ do
+  let inputGenesisShelleyFp = tempAbsPath </> genesisInputFilepath ShelleyEra
+      inputGenesisAlonzoFp  = tempAbsPath </> genesisInputFilepath AlonzoEra
+      inputGenesisConwayFp  = tempAbsPath </> genesisInputFilepath ConwayEra
+
+  -- We write the genesis files to disk, to pass them to create-testnet-data.
```

Comment:
These files shouldn't be modified after they are created. `create-testnet-data` should create the final genesis files we want to use and then we create a configuration file based on those genesis files.

Reply:
Yep agreed, we should remove the last rewrites that remain later on :+1:

Reply:
Can you update the comment so it's obvious this isn't the behaviour we want?